### ISSUE TITLE
divident calculation in networth report

### ISSUE THREAD

**Issue by baartosz** [redstreet/fava-investor#22]
There is no clear indication which stock is generating dividends. Probably they are usualy coming from one income account like `Income:Dividends` or maybe with child accounts per stock. Seems like this query does the job:
```
select sum(cost(position)) where account ~ 'investments' and joinstr(other_accounts) ~ 'dividends'
```
Is there some better way of calculating dividends?

**Comment by redstreet** [comment]
Are you wanting to find all the dividends across the entire portfolio? Or just the dividends corresponding to a single investment account?

In either case, I think your search is a good default, which you can make overridable by the user ([example](https://github.com/redstreet/fava-investor/blob/1c0431b9fc8bbcd915df1b2f0edaaf46eb398e53/fava_investor/modules/tlh/libtlh.py#L141)). I agree joinstr is hacky, but it is the most reasonable default. The alternative is to write code to examine each transaction and pattern match individually on the other accounts. We can do this later if we find we really need it; I suspect we won't.

In general, it is hard to write a search for the general case because it depends on how each person has setup their hierarchy. An example that won't work with the above is when someone sends dividends directly to their Bank (not investment) account.

**Comment by baartosz** [comment]
I gave it some thoughts today morning and it seemed reasonable to split between "reinvested" and "withdrawn" (or "other") dividends depending on where it was transfered. Query above would be for selecting reinvested and then the other group would be difference between sum of dividends and those reinvested. Then we also can have "dividend total" grouping these two. Does it sound okay?

**Comment by redstreet** [comment]
Assuming your goal is "show split of contributions, income (dividends, gains, etc.), and appreciation"  mentioned in the [design doc](https://github.com/redstreet/fava-investor/blob/master/design.md), yes, this seems like a good way to go forward :).

Once you have a PR out, even if it's rough, I can provide more specific feedback.

**Comment by redstreet** [comment]
Feel free to ignore these links for now, but I'm capturing them here for future reference.
Related:
Martin's [returns calculator](https://bitbucket.org/blais/beancount/src/tip/experiments/returns/returns.py) and forum [discussion](https://groups.google.com/forum/#!searchin/beancount/returns.py$20redstreet0%7Csort:date/beancount/XfuetD4BeOY/-lzaPiNqEgAJ).

**Comment by baartosz** [comment]
Interesting read, thanks a lot. Seems like Martin put a lot of effort in it and got quite far. Are you planning to follow these ideas and reuse some of the code for this extension?

**Comment by redstreet** [comment]
Yes, I think that is the right direction for the IRR component of Investor, though I haven't given it much further thought than that.

As you can see, there is some intersection with the net worth component that you're looking into. Net worth can be thought of as having three parts: savings, dividends, and appreciation. It is helpful to be able to see those three parts individually for a portfolio as a whole, and also for any part of a portfolio, whether that be accounts, account groups, or individual commodities.

Being able to separate out dividends is a hard general problem, but I think you are on the right track: I think a reasonable default query like yours is a good place to start, and might get us reasonably close to the finish line.

This is where I
